account_setup.py

```diff
@@ -0,0 +1,578 @@
+        if i != 2:
+            given_tenant_id = input("Input not recognized, please try again: ")
+    raise AccountSetupError("Tenant selection not recognized after 3 attempts.")
+
+
+
+class Spinner:
+    busy = False
+    delay = 0.1
+
+    @staticmethod
+    def spinning_cursor():
+        while 1:
+            for cursor in '|/-\\': yield cursor
+
+    def __init__(self, delay=None):
+        self.spinner_generator = self.spinning_cursor()
+        if delay and float(delay): self.delay = delay
+
+    def __enter__(self):
+        return self.start()
+
+    def __exit__(self, exc_type, exc_val, exc_tb):
+        return self.stop()
+
+    def spinner_task(self):
+        while self.busy:
+            sys.stdout.write(next(self.spinner_generator))
+            sys.stdout.flush()
+            time.sleep(self.delay)
+            sys.stdout.write('\b')
+            sys.stdout.flush()
+
+    def start(self):
+        self.busy = True
+        threading.Thread(target=self.spinner_task, daemon=True).start()
+
+    def stop(self):
+        self.busy = False
+        time.sleep(self.delay)
+
+
+
+if __name__ == "__main__":
+    print("\nGetting credentials.")
+    # get credentials and tenant_id
+    creds, subscription_id = credentials.get_azure_cli_credentials()
+    subscription_client = SubscriptionClient(creds)
+    tenant_ids = [tenant.id for tenant in subscription_client.tenants.list()]
+    if len(tenant_ids) != 1:
+        tenant_id = prompt_tenant_selection(tenant_ids)
+    else:
+        tenant_id = tenant_ids[0]
+
+    if len(sys.argv) > 1 and sys.argv[1] == "deleteresourcegroup":
+        resource_group = input("Resource Group Name: ")
+        with Spinner():
+          delete_resource_group(creds, subscription_id, resource_group)
+        print("Deleted resource group.")
+        sys.exit()
+   
+    if len(sys.argv) > 1 and sys.argv[1] == "serviceprincipal":
+        authentication = "serviceprincipal"
+    else:
+        authentication = DefaultSettings.authentication
+
+    print("Input the desired names and values for your Azure resources. "\
+          "Default values are provided in the brackets. "\
+          "Hit enter to use default.")
+
+    chars = string.ascii_lowercase
+    suffix = "".join(random.choice(chars) for i in range(4))
+    DefaultSettings.storage_account += suffix
+    DefaultSettings.batch_account += suffix
+    if authentication == DefaultSettings.authentication:
+        kwargs = {
+            "region": prompt_with_default("Azure Region", DefaultSettings.region),
+            "resource_group": prompt_with_default("Resource Group Name", DefaultSettings.resource_group),
+            "storage_account": prompt_with_default("Storage Account Name", DefaultSettings.storage_account),
+            "batch_account": prompt_with_default("Batch Account Name", DefaultSettings.batch_account),
+            # "virtual_network_name": prompt_with_default("Virtual Network Name", DefaultSettings.virtual_network_name),
```

Comment:
remove vnet and subnet since they are not valid for shared key auth

Reply:
removed.